language/move-compiler/src/command_line/compiler.rs

```diff
@@ -56,16 +56,18 @@ pub const PASS_PARSER: Pass = 1;
 pub const PASS_EXPANSION: Pass = 2;
 pub const PASS_NAMING: Pass = 3;
 pub const PASS_TYPING: Pass = 4;
-pub const PASS_HLIR: Pass = 5;
-pub const PASS_CFGIR: Pass = 6;
-pub const PASS_COMPILATION: Pass = 7;
+pub const PASS_INLINING: Pass = 5;
+pub const PASS_HLIR: Pass = 6;
+pub const PASS_CFGIR: Pass = 7;
+pub const PASS_COMPILATION: Pass = 8;
 
 #[derive(Debug)]
 enum PassResult {
     Parser(parser::ast::Program),
     Expansion(expansion::ast::Program),
     Naming(naming::ast::Program),
     Typing(typing::ast::Program),
+    Inlining(typing::ast::Program),
```

Comment:
A style consistent to the current compiler implementation would be to define a new `struct inlining::ast::Program` with pretty much everything similar to `struct typing::ast::Program` (and the associated structs) except in `enum Exp` where there is no `Lambda`-related variants.

However, the above style might not be the most time-economical way of implementing a transformation pass (in contrast to a compilation phase). Personally I am fine with either approaches and this PR proved its point on how succinctly this feature can be implemented without duplicating much of data structure (and the associated translation logic).

Reply:
This is a good point. However, I believe even a strict interpretation of this design would still entail to reuse an AST between two phases when there is no significant change in the AST structure (or normal form) of the program. This is the case for inlining. The only difference in the two ASTs is that after the inlining pass, no functions marked as 'inline' will be anymore in the AST. Wondering whether @tnowacki has an opinion here. Still I think I go forward with this.